cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -353,6 +354,12 @@ Result<S3Options> S3Options::FromUri(const Uri& uri, std::string* out_path) {
       options.scheme = kv.second;
     } else if (kv.first == "endpoint_override") {
       options.endpoint_override = kv.second;
+    } else if (kv.first == "allow_bucket_creation") {
+      options.allow_bucket_creation =
+          ::arrow::internal::AsciiEqualsCaseInsensitive(kv.second, "true");
+    } else if (kv.first == "allow_bucket_deletion") {
+      options.allow_bucket_deletion =
+          ::arrow::internal::AsciiEqualsCaseInsensitive(kv.second, "true");
```

Comment:
Hmm, I think we should be a bit stricter and not blindly interpret any other value as "false".
How about allowing the following values:
* "0", "false" (case-insensitive) -> boolean false
* "1", "true" (case-insensitive) -> boolean true
* any other value -> raise `Status::Invalid`

Reply:
Sounds good. I didn't see any existing utility function to do this, so maybe I will create one.